app/presenters/publication_presenter.rb

```diff
@@ -129,9 +129,46 @@ def video_embed_url
     end
   end
 
+  def promotion_choice
+    choice = promotion_choice_details['choice']
+    if choice.empty?
+      if has_legacy_organ_promotion_details?
+        "organ_donor"
+      else
+        "none"
+      end
+    else
+      choice
+    end
+  end
+
+  def promotion_url
+    url = promotion_choice_details['url']
+    url.empty? ? legacy_organ_promotion_details['organ_donor_registration_url'] : url
+  end
```

Comment:
After discussion what we think is a good solution is:

1. deploy these complicated versions of frontend and publisher that support both old and new style of promo storage
2. deploy a data-migration that re-writes promo storage for all documents into new style only
3. deploy new simpler versions of frontend and publisher that only support new style of promo storage

Note that 2 and 3 are, as yet, unwritten.